src/components/layout/BaseLayout.tsx

```diff
@@ -1,20 +1,63 @@
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
 import { THEME } from "config/theme";
 import { pxValue } from "helpers/dimensions";
 import React from "react";
+import {
+  KeyboardAvoidingView,
+  KeyboardAvoidingViewProps,
+  ScrollView,
+  ScrollViewProps,
+  Platform,
+} from "react-native";
 import { EdgeInsets, useSafeAreaInsets } from "react-native-safe-area-context";
 import styled from "styled-components/native";
 
 interface BaseLayoutProps {
   children: React.ReactNode;
   useSafeArea?: boolean;
   backgroundColor?: string;
+  useKeyboardAvoidingView?: boolean;
 }
 
 interface StyledViewProps {
   $backgroundColor: string;
   $insets: EdgeInsets;
 }
 
+const StyledKeyboardAvoidingView = styled(KeyboardAvoidingView).attrs(
+  (props: KeyboardAvoidingViewProps) => ({
+    behavior: Platform.select({
+      ios: "padding",
+      android: undefined,
+    }),
+    contentContainerStyle: {
+      flex: 1,
+      backgroundColor: THEME.colors.background.default,
+    },
+    ...props,
+  }),
+)`
+  flex: 1;
+  background-color: ${THEME.colors.background.default};
+`;
+
+const StyledScrollView = styled(ScrollView).attrs(
+  (props: ScrollViewProps) =>
+    ({
+      keyboardShouldPersistTaps: "never",
+      showsVerticalScrollIndicator: false,
+      alwaysBounceVertical: false,
+      contentContainerStyle: {
+        flex: 1,
+        backgroundColor: THEME.colors.background.default,
+      },
+      ...props,
+    }) as ScrollViewProps,
+)`
+  flex-grow: 1;
+  background-color: ${THEME.colors.background.default};
+`;
+
```

Comment:
It appears this is a copy/paste of what we have on `OnboardLayout`. Could we extract it to a reusable component?